applications/StructuralMechanicsApplication/custom_strategies/custom_schemes/explicit_central_differences_scheme.hpp

```diff
@@ -0,0 +1,900 @@
+      {
+        typename NodesArrayType::iterator i_begin = r_nodes.ptr_begin()+node_partition[k];
+        typename NodesArrayType::iterator i_end   = r_nodes.ptr_begin()+node_partition[k+1];
+
+        for(ModelPart::NodeIterator i=i_begin; i!= i_end; ++i)
+        {
+          array_1d<double,3>& r_middle_velocity       = i->GetValue(MIDDLE_VELOCITY);
+          array_1d<double,3>& r_current_velocity      = i->FastGetSolutionStepValue(VELOCITY);
+          array_1d<double,3>& r_current_residual      = i->FastGetSolutionStepValue(FORCE_RESIDUAL);
+          //array_1d<double,3>& r_current_displacement  = i->FastGetSolutionStepValue(DISPLACEMENT);
+          
+          for (unsigned int j =0; j<3; j++)
+          {
+            
+            r_middle_velocity[j]      = r_current_velocity[j] ;
+            r_current_residual[j]     = 0.0;
+            //r_current_displacement[j] = 0.0; // this might be wrong for presribed displacement
+          }
+
+          if (i->HasDofFor(ROTATION_X))
+          {
+            array_1d<double,3>& r_middle_angular_velocity       = i->GetValue(MIDDLE_ANGULAR_VELOCITY);
+            array_1d<double,3>& r_current_angular_velocity      = i->FastGetSolutionStepValue(ANGULAR_VELOCITY);
+            array_1d<double,3>& r_current_residual_moment       = i->FastGetSolutionStepValue(MOMENT_RESIDUAL);
+            //array_1d<double,3>& current_rotation              = i->FastGetSolutionStepValue(ROTATION);    
+            
+            for (unsigned int j =0; j<3; j++)
+            {
+              
+              r_middle_angular_velocity[j]      = r_current_angular_velocity[j] ;
+              r_current_residual_moment[j]     = 0.0;
+              //current_rotation[j] = 0.0; // this might be wrong for presribed rotations
+            }
+          }
+
+
+
+        }
+      }
+    KRATOS_CATCH("")
+	  }
+
+  /**
+      Performing the update of the solution.
+  */
+  //***************************************************************************
+    virtual void Update(ModelPart& rModelPart,
+          DofsArrayType& rDofSet,
+          TSystemMatrixType& A,
+          TSystemVectorType& Dx,
+          TSystemVectorType& b
+          )
+    {
+      KRATOS_TRY
+      ProcessInfo& r_current_process_info  = rModelPart.GetProcessInfo();
+      NodesArrayType& r_nodes            = rModelPart.Nodes();
+      const double numerical_limit = std::numeric_limits<double>::epsilon();
+      //Step Update
+      mTime.Current   = r_current_process_info[TIME];  //the first step is time = initial_time ( 0.0) + delta time
+      mTime.Delta     = r_current_process_info[DELTA_TIME];
+
+      mTime.Middle    = 0.5*(mTime.Previous + mTime.Current);
+
+
+      #ifdef _OPENMP
+        int number_of_threads = omp_get_max_threads();
+      #else
+        int number_of_threads = 1;
+      #endif
+
+      vector<unsigned int> node_partition;
+      OpenMPUtils::CreatePartition(number_of_threads, r_nodes.size(), node_partition);
```

Comment:
avoid partitioning explicitly. Have OpenMP to do it for you.
(i am aware it is there in some existing code...however it is most probably less efficient than relying on openmp)

Reply:
fixed